engine/wwtlib/WWTControl.cs

```diff
@@ -28,12 +28,28 @@ static WWTControl()
             SpaceTimeController.UpdateClock();
         }
 
-        public static List<Imageset> ImageSets = new List<Imageset>();
+        private static List<Imageset> ImageSets = new List<Imageset>();
         public static Folder ExploreRoot = new Folder();
         public static string ImageSetName = "";
 
         public IUiController uiController = null;
 
+        public static void AddImageSet(Imageset imagesetToAdd)
+        {
+            foreach(Imageset imageset in ImageSets){
+                if(imageset.ImageSetID == imagesetToAdd.ImageSetID)
```

Comment:
What's the rationale for adding this rejection of duplicates? And, should it be using the URL instead of the ImageSetId? As far as I can tell the ImageSetID is based on a weak hash of the URL, so I'm not sure if it adds any value.

Reply:
I only see problems with having duplicates. Especially when there are settings like FITS cuts attached to an image set. 
For example, changing the duplicated imagesets cuts would not reflect in the imageset that is actually showing. 

Yeah, using the URL instad is fine

Reply:
OK. I tend to think that we should enforce URL-based uniqueness. I do worry about possible cases where the same image-data URL is associated with different overall Imageset objects. Two examples I can think of:

- There might be uses cases to use the a kind of generic "always give me a black tile" URL for multiple different imagesets
- You might want to imagesets with the same "study" image data to compare different astrometric solutions

But, I do see a lot of value in being able to uniquely identify imagesets by URL, these two examples are definitely corner cases, and you can always uniquify a URL by adding an unused query parameter or fragment.

Reply:
Hmm, there is also the case of HiPS, with multiple image formats. Depending on how we deal with it there could be multiple imagesets with the same URL and different extensions (FITS, jpg, png)

Reply:
Ah, I see what you're saying. I think that it would be not-unreasonable to use a URL fragment to disambiguate if needed? E.g. `http://.../my/hips#fits` vs. `http://.../my/hips#jpg`.

Reply:
Yes, something like that could be implemented for HiPS.

At least the ImageSetID has to be unique, since that is currently used to find and access tiles in the TileCache. Imagesets with the same URL will now refer to the exact same Tiles in the draw cycle.
Of course that could be solved by using a composite of traits to create the ID or some other way to create a ID for each imageset.